File: tournaments/digitDegree/digitDegree.cs
int digitDegree(int n) {
    var result = 0;

    while (10 <= n) {
        n = getSumOfDigits(n);
        ++result;
    }

    return result;
}

int getSumOfDigits(int n) {
    var result = 0;

    while (0 < n) {
        result += n % 10;
        n /= 10;
    }

    return result;
}

File: tournaments/digitDegree/digitDegree.js
function digitDegree(n) {
    const getSumOfDigits = function(a) {
        let r = 0;

        while (0 < a) {
            r += a % 10;
            a = ~~(a / 10);
        }

        return r;
    };

    let result = 0;

    while (10 <= n) {
        n = getSumOfDigits(n);
        ++result;
    }

    return result;
}
